news/_posts/2016-10-18-release-notes-2.12.0-RC2.md

```diff
@@ -0,0 +1,198 @@
+
+For 2.12.0, we will try to remain binary compatible with RC1,
+and we won't risk regressions except for the most critical bugs.
+
+#### Trait compiles to an interface
+
+With Java 8 allowing concrete methods in interfaces, Scala 2.12 is able to compile a trait to a single interface. Before, a trait was represented as a class that held the method implementations and an interface. Note that the compiler still has quite a bit of magic to perform behind the scenes, so that care must be taken if a trait is meant to be implemented in Java. (Briefly, if a trait does any of the following its subclasses require synthetic code: defining fields, calling super, initializer statements in the body, extending a class, relying on linearization to find implementations in the right super trait.)
+
+#### Java 8-style lambdas
+
+Scala 2.12 emits closures in the same style as Java 8, whether they target a `FunctionN` class from the standard library or a user-defined Single Abstract Method (SAM) type. The type checker accepts a function literal as a valid expression for either kind of "function-like" type (built-in or SAM). This improves the experience of using libraries written for Java 8 in Scala.
+
+For each lambda the compiler generates a method containing the lambda body, and emits an `invokedynamic` that will spin up a lightweight class for this closure using the JDK's `LambdaMetaFactory`.
+
+Compared to Scala 2.11, the new scheme has the advantage that, in most cases, the compiler does not need to generate an anonymous class for each closure. This leads to significantly smaller JAR files.
+
+#### New back end
+
+Scala 2.12 standardizes on the "GenBCode" back end, which emits code more quickly because it directly generates ASM bytecode from Scala compiler trees, while the previous back end used an intermediate representation called "ICode". The old back ends (GenASM and GenIcode) have been removed ([#4814](https://github.com/scala/scala/pull/4814), [#4838](https://github.com/scala/scala/pull/4838)).
+
+#### New optimizer
+
+The GenBCode back end includes a new inliner and bytecode optimizer.
+The optimizer is enabled using `-opt` compiler option, which defaults
+to `-opt:l:classpath`.  Check `-opt:help` to see the full list of
+available options for the optimizer.
+
+The following optimizations are available:
+
+* Inlining final methods, including methods defined in objects and final methods defined in traits
+* If a closure is allocated and invoked within the same method, the closure invocation is replaced by an invocations of the corresponding lambda body method
+* Dead code elimination and a small number of cleanup optimizations
+* Box/unbox elimination [#4858](https://github.com/scala/scala/pull/4858)
+
+#### Either is now right-biased
+
+`Either` now supports operations like `map`, `flatMap`, `contains`,
+`toOption`, and so forth, which operate on the right-hand side.
+
+(`.left` and `.right` will be deprecated in favor of `.swap` in a later release.)
```

Comment:
s/will be/may be/ — I think `.right` will definitely be deprecated but the situation with `.left` is less clear